src/lib/flatten/tsc.ts

```diff
@@ -1,48 +1,29 @@
 import * as path from 'path';
+import { SourceDescription } from 'rollup';
 import { ScriptTarget, ModuleKind, TranspileOutput, transpileModule, CompilerOptions } from 'typescript';
-import { readFile, outputJson, outputFile as fsOutputFile } from 'fs-extra';
 import { debug } from '../util/log';
 
 /**
  * Downlevels a .js file from ES2015 to ES5. Internally, uses `tsc`.
  *
- * @param inputFile
- * @param outputFile
  */
-export async function downlevelWithTsc(inputFile: string, outputFile: string): Promise<void> {
-  debug(`tsc ${inputFile} to ${outputFile}`);
-  const inputBuffer: Buffer = await readFile(inputFile);
-  const input: string = inputBuffer.toString();
+export async function downlevelWithTsc(code: string, filePath: string): Promise<SourceDescription> {
+  debug(`tsc ${filePath}`);
   const compilerOptions: CompilerOptions = {
     target: ScriptTarget.ES5,
     module: ModuleKind.ES2015,
     allowJs: true,
```

Comment:
I am not sure, why this is needed, in general, libraries that generate `.d.ts` cannot use this.

Reply:
What do you mean "this"?

Reply:
I am not sure what you refer to with "this". Is it the sourcemaps? I remember that was needed as it was transpiling the fesm15 file and w/o stripping it had two "// sourceMappingUrl" lines at end of fesm5 file.

Reply:
This was just as a side note. `allowJs` usually shouldn't be enabled. for libraries as libraries usually generate `d.ts` with `declaration: true` and this flag doesn't work with `declaration: true`.

So in theory I don't see the point of having `allowJs` in a library that generated `d.ts`

Reply:
`allowJs` is for adding `.js` files to the compilation as we're down-levelling from a `*.js` file (not a `*.ts.`)

> Allow JavaScript files to be compiled.

https://www.typescriptlang.org/docs/handbook/compiler-options.html

Reply:
Ah thanks mate for the info